### PR TITLE
[RND-543] Run Meadowlark in Containers on Azure

### PR DESCRIPTION
**Pull request by andonyns**
- Adding instructions to deploy to azure.
- Adding docker compose file to simplify deployment.
- Adding meadowlark-opensearch image to extract env variables

### CODE REVIEW

File: Meadowlark-js/deploys/.env.example
[the remark this answers is not in the document]
Comment (andonyns) [reply Meadowlark-js/deploys/.env.example:1]: I like this, I will do the change

File: eng/deploy/azure/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+```Shell
+docker context use myacicontext
+```
+
+- Login to Azure Container Registry
+
+```Shell
+az acr login --name edfimeadowlark
+```
```
Comment (stephenfuqua) [inline eng/deploy/azure/README.md:26]: I was thinking something more like this, except you would need to add the rest of the commands into this same Shell block.

```suggestion
- [Log into Azure from
  Docker](https://docs.docker.com/cloud/aci-integration/#log-into-azure).
- [Create an ACI Docker
  context](https://docs.docker.com/cloud/aci-integration/#create-an-aci-context)
- Switch to the ACI Context and Deploy Meadowlark

   ```Shell
   docker context use myacicontext
   az acr login --name edfimeadowlark
   ````
```

File: eng/deploy/azure/README.md [hunk 2]
```diff
@@ -13,79 +13,58 @@ CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli).
...
-```Shell
 docker compose -p meadowlark --file ./azure-docker-compose.yml up -d
-```
-
-- Initialize mongodb replica set
 
-```Shell
+# Initialize replica set
 az container exec --resource-group {resource group name} -n meadowlark
 --container-name ml-mongo1 --exec-command 'mongo --eval rs.initiate()'
```
Comment (stephenfuqua) [inline eng/deploy/azure/README.md]: Won't this fail without a line continuation character?
```suggestion
az container exec --resource-group {resource group name} -n meadowlark `
    --container-name ml-mongo1 --exec-command 'mongo --eval rs.initiate()'
```

Annoying thing is that I entered the PowerShell line continuation, which won't work for Bash. And vice versa. I'm OK with that though.

File: eng/deploy/azure/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+docker context use myacicontext
+
+# Login to Azure Container Registry
+az acr login --name edfimeadowlark
+
+docker compose -p meadowlark --file ./azure-docker-compose.yml up -d
+
+# Initialize replica set
+az container exec --resource-group {resource group name} -n meadowlark
+--container-name ml-mongo1 --exec-command 'mongo --eval rs.initiate()'
```
Comment (stephenfuqua) [inline eng/deploy/azure/README.md:34]: Needs a line continuation character:

```suggestion
az container exec --resource-group {resource group name} -n meadowlark `
   --container-name ml-mongo1 --exec-command 'mongo --eval rs.initiate()'
```

File: eng/deploy/azure/README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+# Create OpenSearch container
+az container create --resource-group {resource group name} -n meadowlark
+--image edfimeadowlark.azurecr.io/meadowlark-opensearch:latest --dns-name-label ml-opensearch1
+
+# Create meadowlark container
+az container create --resource-group {resource group name} -n meadowlark
+--image edfialliance/meadowlark-ed-fi-api:pre --ports 80 --environment-variables {specify all env variables required}
+
+
```
Comment (stephenfuqua) [inline eng/deploy/azure/README.md:67]:
```suggestion
az container create --resource-group {resource group name} -n ml-mongo `
    --image mongo:4.0.28 --ports 27017 --dns-name-label mlmongo1 `
    --command-line "mongod --replSet rs0"

# Initialize mongodb replica set
az container exec --resource-group {resource group name} -n meadowlark `
    --container-name ml-mongo1 --exec-command 'mongo --eval rs.initiate()'

# Create OpenSearch container
az container create --resource-group {resource group name} -n meadowlark `
    --image edfimeadowlark.azurecr.io/meadowlark-opensearch:latest
    --dns-name-label ml-opensearch1

# Create meadowlark container
az container create --resource-group {resource group name} -n meadowlark `
    --image edfialliance/meadowlark-ed-fi-api:pre --ports 80 `
    --environment-variables {specify all env variables required}
```

**Comment by stephenfuqua** [comment]
This is a good work, though I wasn't able to get a Meadowlark environment working correctly with the `az` commands. This ticket was a spike. We will have a follow-up ticket for creating the fully clean and polished deployment process. Therefore I'm going to close this ticket without deleting the branch, and I'm going to "rename" the branch as `spike-azure-may2023`.

**Pull request closed without merging**